src/features/game/dialogue/GameDialogueManager.ts

```diff
@@ -102,8 +105,43 @@ export default class DialogueManager {
     }
   }
 
+  /**
+   * Hide all dialogue boxes, speaker boxes and speaker sprites
+   * */
+  public async hideAll() {
+    await this.getDialogueRenderer().hide();
+    await this.getSpeakerRenderer().hide();
+  }
+
+  /**
+   * Make all dialogue boxes, speaker boxes and speaker sprites visible
+   * */
+  public async showAll() {
+    await this.getDialogueRenderer().show();
+    await this.getSpeakerRenderer().show();
+  }
+
+  /**
+   * Find patterns of quiz score interpolation in a dialogue line,
+   * and replace them by actual scores.
+   * The pattern: "{<quizId>.score}"
+   *
+   * @param line
+   * @returns {string} the given line with all quiz score interpolation replaced by actual scores.
+   */
+  public makeLineWithQuizScores(line: string) {
+    const quizScores = line.match(/\{.+?\.score\}/g);
+    if (quizScores) {
+      quizScores.forEach(match => {
+        const quizId = match.substring(1, match.lastIndexOf('.'));
+        line = line.replace(match, GameGlobalAPI.getInstance().getQuizScore(quizId).toString());
+      });
+    }
+    return line;
+  }
+
   private getDialogueGenerator = () => this.dialogueGenerator as DialogueGenerator;
-  private getDialogueRenderer = () => this.dialogueRenderer as DialogueRenderer;
+  public getDialogueRenderer = () => this.dialogueRenderer as DialogueRenderer;
```

Comment:
Is there a need to change this from private to public?

Reply:
Comment addressed: commit 6953459107c1578548597535bf47d0b13cecf7cc